app/ante/cosmos/vesting.go

```diff
@@ -13,85 +13,19 @@
 	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	errorsmod "cosmossdk.io/errors"
 	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/x/authz"
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 	evmtypes "github.com/evmos/evmos/v11/x/evm/types"
 	vestingtypes "github.com/evmos/evmos/v11/x/vesting/types"
 )
 
-// EthVestingTransactionDecorator validates if clawback vesting accounts are
-// permitted to perform Ethereum Tx.
-type EthVestingTransactionDecorator struct {
-	ak evmtypes.AccountKeeper
-}
-
-func NewEthVestingTransactionDecorator(ak evmtypes.AccountKeeper) EthVestingTransactionDecorator {
-	return EthVestingTransactionDecorator{
-		ak: ak,
-	}
-}
-
-// AnteHandle validates that a clawback vesting account has surpassed the
-// vesting cliff and lockup period.
-//
-// This AnteHandler decorator will fail if:
-//   - the message is not a MsgEthereumTx
-//   - sender account cannot be found
-//   - sender account is not a ClawbackvestingAccount
-//   - blocktime is before surpassing vesting cliff end (with zero vested coins) AND
-//   - blocktime is before surpassing all lockup periods (with non-zero locked coins)
-func (vtd EthVestingTransactionDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
-	for _, msg := range tx.GetMsgs() {
-		msgEthTx, ok := msg.(*evmtypes.MsgEthereumTx)
-		if !ok {
-			return ctx, errorsmod.Wrapf(errortypes.ErrUnknownRequest,
-				"invalid message type %T, expected %T", msg, (*evmtypes.MsgEthereumTx)(nil),
-			)
-		}
-
-		acc := vtd.ak.GetAccount(ctx, msgEthTx.GetFrom())
-		if acc == nil {
-			return ctx, errorsmod.Wrapf(errortypes.ErrUnknownAddress,
-				"account %s does not exist", acc)
-		}
-
-		// Check that this decorator only applies to clawback vesting accounts
-		clawbackAccount, isClawback := acc.(*vestingtypes.ClawbackVestingAccount)
-		if !isClawback {
-			return next(ctx, tx, simulate)
-		}
-
-		// Error if vesting cliff has not passed (with zero vested coins). This
-		// rule does not apply for existing clawback accounts that receive a new
-		// grant while there are already vested coins on the account.
-		vested := clawbackAccount.GetVestedCoins(ctx.BlockTime())
-		if len(vested) == 0 {
-			return ctx, errorsmod.Wrapf(vestingtypes.ErrInsufficientVestedCoins,
-				"cannot perform Ethereum tx with clawback vesting account, that has no vested coins: %s", vested,
-			)
-		}
-
-		// Error if account has locked coins (before surpassing all lockup periods)
-		islocked := clawbackAccount.HasLockedCoins(ctx.BlockTime())
-		if islocked {
-			return ctx, errorsmod.Wrapf(vestingtypes.ErrVestingLockup,
-				"cannot perform Ethereum tx with clawback vesting account, that has locked coins: %s", vested,
-			)
-		}
-	}
-
-	return next(ctx, tx, simulate)
-}
-
 // TODO: remove once Cosmos SDK is upgraded to v0.46
 
 // VestingDelegationDecorator validates delegation of vested coins
```

Comment:
@fedekunze @facs95  Do we still need this decorator?

Reply:
let's keep it as is for now